scheduler/src/cook/scheduler/constraints.clj

```diff
@@ -177,11 +177,57 @@
     (when (and (not (empty? datasets)) (= fitness-calculator dl/data-local-fitness-calculator))
       (->data-locality-constraint job launch-wait-seconds))))
 
-(defrecord user-defined-constraint [constraints]
+(defn job->default-constraints
+  "Returns the list of default constraints configured for the job's pool"
+  [job]
+  (util/match-based-on-pool-name
+    (config/default-job-constraints)
+    (util/job->pool-name job)
+    :default-constraints))
+
+(def machine-type-constraint-attributes
+  #{"cpu-architecture" "node-family" "node-type"})
+
+(defn job->constraints
```

Comment:
I think we should de-lazy this output; dump into a vector or list. Fenzo can call from multiple threads.

Reply:
Will do, thanks

Reply:
Done